File: springboot/src/main/java/com/example/muscle_api/entity/Record.java
package com.example.muscle_api.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Record {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate date;

    private String exercise;

    private int weight;

    private int reps;

    private int sets;

    private String memo;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user; // ユーザーエンティティとのリレーション

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: flutter/lib/models/record.dart
class Record {
  final int id;
  final String date;
  final String exercise;
  final int weight;
  final int reps;
  final int sets;
  final String? memo; // ✅ ここが null を許容していること

  Record({
    required this.id,
    required this.date,
    required this.exercise,
    required this.weight,
    required this.reps,
    required this.sets,
    this.memo,
  });

  factory Record.fromJson(Map<String, dynamic> json) {
    return Record(
      id: json['id'],
      date: json['date'] ?? '', // ✅ 念のため fallback を追加しても良い
      exercise: json['exercise'] ?? '',
      weight: json['weight'] ?? 0,
      reps: json['reps'] ?? 0,
      sets: json['sets'] ?? 0,
      memo: json['memo'],
    );
  }
}
